src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_Fields.cs

```diff
@@ -0,0 +1,608 @@
+            comp.VerifyDiagnostics(
+                // (6,14): error CS0843: Auto-implemented property 'S.P' must be fully assigned before control is returned to the caller.
+                //     internal S(string s)
+                Diagnostic(ErrorCode.ERR_UnassignedThisAutoProperty, "S").WithArguments("S.P").WithLocation(6, 14),
+                // (6,14): error CS0171: Field 'S.F' must be fully assigned before control is returned to the caller
+                //     internal S(string s)
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "S").WithArguments("S.F").WithLocation(6, 14),
+                // (6,14): warning CS8618: Non-nullable property 'P' is uninitialized.
+                //     internal S(string s)
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "S").WithArguments("property", "P").WithLocation(6, 14),
+                // (6,14): warning CS8618: Non-nullable field 'F' is uninitialized.
+                //     internal S(string s)
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "S").WithArguments("field", "F").WithLocation(6, 14));
+        }
+
+        [Fact]
+        public void EmptyStruct()
+        {
+            var source =
+@"struct S
+{
+    S(object o)
+    {
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
+        }
+
+        // PROTOTYPE(NullableReferenceTypes): Struct assign `this`.
+        [Fact(Skip = "Assign this")]
+        public void StructAssignThis()
+        {
+            var source =
+@"struct S
+{
+#pragma warning disable 0169
+    private readonly string F;
+    private string[] P { get; set; }
+    internal S(S s)
+    {
+        this = s;
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void StructObjectCreation()
+        {
+            var source =
+@"#pragma warning disable 0649
+struct S
+{
+    internal string F;
+    internal string G;
+}
+class C
+{
+    static void F(S s) { }
+    static void Main()
+    {
+        F(new S());
+        F(new S() { F = string.Empty });
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ValueTypeFields()
+        {
+            var source =
+@"#pragma warning disable 0169
+struct S { }
+class C
+{
+    private readonly S s;
+    private int i;
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TryCatch()
+        {
+            // PROTOTYPE(NullableReferenceTypes):
+        }
+
+        [Fact]
+        public void LocalFunction()
```

Comment:
Just to clarify what you're tracking as follow-up: do you mean to test a constructor with a local function here?

Reply:
Yes, perhaps something like:
```
class C
{
    private object F;
    C()
    {
        void L(object o)
        {
            F = o;
        }
        L(new object());
    }
}
```

Reply:
Added test.